File: simple-github/src/main/java/com/androidhuman/example/simplegithub/data/AuthTokenProvider.java
package com.androidhuman.example.simplegithub.data;

import android.content.Context;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public final class AuthTokenProvider {

    private static final String KEY_AUTH_TOKEN = "auth_token";

    private Context context;

    public AuthTokenProvider(@NonNull Context context) {
        this.context = context;
    }

    public void updateToken(@NonNull String token) {
        PreferenceManager.getDefaultSharedPreferences(context).edit()
                .putString(KEY_AUTH_TOKEN, token)
                .apply();
    }

    @Nullable
    public String getToken() {
        return PreferenceManager.getDefaultSharedPreferences(context)
                .getString(KEY_AUTH_TOKEN, null);
    }

}

File: simple-github/src/main/kotlin/com/androidhuman/example/simplegithub/data/AuthTokenProvider.kt
package com.androidhuman.example.simplegithub.data

import android.content.Context
import android.preference.PreferenceManager

private const val KEY_AUTH_TOKEN = "auth_token"

class AuthTokenProvider(private val context: Context) {

    fun updateToken(token: String) {
        PreferenceManager.getDefaultSharedPreferences(context).edit()
                .putString(KEY_AUTH_TOKEN, token)
                .apply()
    }

    val token: String?
        get() = PreferenceManager.getDefaultSharedPreferences(context)
                .getString(KEY_AUTH_TOKEN, null)
}
